server.go

```diff
@@ -45,6 +45,13 @@ type Server struct {
 	wg           sync.WaitGroup
 	timeout      time.Duration
 	retries      int
+	sendAEnable  bool /* senderAnticipate enable by server */
+	sendAWinSz   uint
+}
+
+func (s *Server) SetAnticipate(en bool, winsz uint) {
```

Comment:
Could it be `SetAnticipate(windowSize uint)`? Then enable `senderAnticipate` in case window size is larger than 1? So that `SetAnticipate(0)` or `SetAnticipate(1)` disables the feature, and `SetAnticipate(5)` enables it.

Motivation: avoid nonsense parameter combos such as `SetAnticipate(false, 42)`

I need to look more carefully at what window size means exactly - just glanced over the sender code. Window size of 1 is standard TFTP operation basically, am I getting it right? If that right, `SetAnticipate(0)` could even actually panic as it does not make sense?

Reply:
Sounds sensible. Will get back to it in a few days or so.